cloudify_agent/api/factory.py

```diff
@@ -197,16 +197,21 @@ def save(self, daemon):
             self.storage, '{0}.json'.format(
                 daemon.name)
         )
+
+        if getattr(daemon, 'cluster'):
+            cluster_settings.set_cluster_nodes(daemon.cluster)
+            daemon.cluster = cluster_settings.get_cluster_nodes()
```

Comment:
first `set_cluster_nodes` so that the mechanism there can normalize the node dicts; then get it back again, so we can stored the normalized dicts on `daemon.cluster`